### DISCUSSION TITLE
Dynamic text inputs

### DISCUSSION THREAD

**Discussion by jelmer-devries** [pimcore/pimcore#7896] [Q&A / Support] [answered]
Hi Pimcore developers,

Maybe my question seems weird, but correct me on on that one if that's the case.

How'd we create dynamic text inputs for an object class? For example we need to have Pro's and con's. For one product it'd be 2 pro's and 1 con, but for an other project it could be like 5 and 4. How do we create dynamic text inputs, so they can click on some button to generate a new text field in the data object and fill it in. Or do we have to use the relations for this? Which seems a bit overkill to me.

**Comment by NiklasBr** [comment, marked as answer]
Blocks should be your answer here, they can be repeated arbitrarily (or with a limit) and contain a number of different data fields:
![image](https://user-images.githubusercontent.com/279826/104313217-00eaf780-54d8-11eb-87ba-191ce3cf4118.png)

**Reply by jelmer-devries** [reply]
Thank you so much :_)
